backend/src/libs/packages/controller/libs/types/api-handler-options.type.ts

```diff
@@ -6,9 +8,13 @@ type DefaultApiHandlerOptions = {
   params?: unknown;
 };
 
+type HeadersInjected = {
+  headers: FastifyRequest['headers'];
+};
+
 type ApiHandlerOptions<
   T extends DefaultApiHandlerOptions = DefaultApiHandlerOptions,
-> = {
+> = HeadersInjected & {
   body: T['body'];
```

Comment:
```suggestion
> = {
  headers: type
  body: T['body'];
```

Reply:
prefer `Parameters<ServerAppRouteParameters>`

Reply:
when I use headers as Parameters<ServerAppRouteParameters>, in mapRequest of controller package it's passing as IncomingHttpHeaders. I received this error in controller package: Property 'headers' is missing in type 'IncomingHttpHeaders' but required in type 'HeadersInjected'.